docs/src/main/asciidoc/spanner.adoc

```diff
@@ -25,55 +27,294 @@ dependencies {
 }
 ----
 
+This setup takes care of bringing in the latest compatible version of Cloud Java Spanner libraries as well.
+
+
+=== Configuration
+
+To setup Spring Data Spanner, you have to configure the following:
+
+* setup the connection details to Google Cloud Spanner
+* enable Spring Data Repositories (optional).
+
+==== Spanner settings
 
-=== Google Cloud Spanner
+You can the use link:../spring-cloud-gcp-starters/spring-cloud-gcp-starter-data-spanner[Spring Boot Starter for Spring Data Spanner] to autoconfigure Google Cloud Spanner in your Spring application.
+It contains all the necessary setup that makes it easy to authenticate with your Google Cloud project.
+The following configuration options are available:
+
+|===
+| Name | Description | Optional | Default value
+| `spring.cloud.gcp.spanner.instance-id` | Spanner instance to use | No |
+| `spring.cloud.gcp.spanner.database` |
+Spanner database to use | No |
+| `spring.cloud.gcp.spanner.project-id` | GCP project ID where the Google Cloud Spanner API
+is hosted, if different from the one in the <<spring-cloud-gcp-core,Spring Cloud GCP Core Module>>
+| Yes |
+| `spring.cloud.gcp.spanner.credentials.location` | OAuth2 credentials for authenticating with the
+Google Cloud Spanner API, if different from the ones in the
+<<spring-cloud-gcp-core,Spring Cloud GCP Core Module>> | Yes |
+| `spring.cloud.gcp.spanner.credentials.scopes` |
+https://developers.google.com/identity/protocols/googlescopes[OAuth2 scope] for Spring Cloud GCP
+Spanner credentials | Yes | https://www.googleapis.com/auth/spanner.data
+|===
 
-Spring Data Spanner allows you to define entity classes and corresponding repository interfaces:
+==== Repository settings
+
+Spring Data Repositories can be configured via the `@EnableSpannerRepositories` annotation on your main `@Configuration` class.
+With our Spring Boot Starter for Spring Data Spanner, `@EnableSpannerRepositories` is automatically added, it is not required to add it to any other class, unless there is a need to override finer grain configuration parameters provided by https://github.com/spring-cloud/spring-cloud-gcp/blob/master/spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/config/EnableSpannerRepositories.java[`@EnableSpannerRepositories`].
+
+==== Autoconfiguration
+
+Our Spring Boot autoconfiguration creates the following beans available in the Spring application context:
+
+- an instance of `SpannerTemplate`,
```

Comment:
there's a comma at the end of this one, but none of the other bullets in this little list